bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -91,7 +91,9 @@ data:
     {{- end }}
     export EXTERNAL_ACCESS_PORT={{ .Values.externalAccess.service.port }}
     {{- else if eq .Values.externalAccess.service.type "NodePort" }}
-    {{- if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
+    {{- if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
+    export EXTERNAL_ACCESS_IP="${MY_POD_IP}"
+    {{- else if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
     export EXTERNAL_ACCESS_IP="${HOST_IP}"
     {{- else if .Values.externalAccess.service.domain }}
```

Comment:
I have updated the logic here. The previous updated,

```
    {{- if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
    export EXTERNAL_ACCESS_IP="${HOST_IP}"
    {{- else if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
    export EXTERNAL_ACCESS_IP="${MY_POD_IP}"
```
In case, `.Values.externalAccess.autoDiscovery.enabled ` is true, the logic will never go to the `else if` statement because it is using `or` condition.

It should to change the logic as below to make sure that in case we enable usePodIPs, it will do that. The new logic
```
    {{- if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
    export EXTERNAL_ACCESS_IP="${MY_POD_IP}"
    {{- else if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
    export EXTERNAL_ACCESS_IP="${HOST_IP}"
```
It only use the MY_POD_IP when both flags: usePodIPs, autoDiscovery.enabled is true. For other cases, it will use HOST_IP.